File: linear_algebra/Python/HHP_BT9.py
import numpy as np

m = int(input("Nhập số hàng: "))
n = int(input("Nhập số cột: "))
A = np.zeros((m,n))
for i in range (m):
    for j in range(n):
      A[i][j] = int(input(f"Nhập phần tử A[{i}][{j}]: "))
print(A)
print()
if m!=n:
   print("Ma trận này không phải là ma trận vuông")
else:
   duongcheo = True
   tamgiactren =True
   tamgiacduoi = True
   doixung = True
   for i in range(m):
      for j in range(n):
        if i != j and A[i][j]!=0:
            duongcheo = False
        if i > j and A[i][j] != 0:
            tamgiactren = False
        if i < j and A[i][j] != 0:
           tamgiacduoi = False
        if A[i][j]!=A[j][i]:
           doixung = False
if duongcheo:
    print("Đây là ma trận đường chéo")
elif tamgiactren:
      print("Đây là ma trận tam giác trên")
elif tamgiacduoi:
      print("Đây là ma trận tam giác dưới")
elif doixung:
      print("Đây là ma trận đối xứng")
else:
      print("Đây không phải là ma trận thuộc các loại trên!!")

File: linear_algebra/C++/HHP_BT9.cpp
#include<iostream>
using namespace std;

int main()
{
    int m, n;
    cout << "Nhap so hang: ";
    cin >> m ;
    cout << "Nhap so cot: ";
    cin >> n;
    double A[m][n];
    for (int i = 0; i < m; i++)
    {
        for (int j = 0 ; j < n; j++)
        {
            cout << "Nhap phan tu A[" << i << "][" << j << "]: ";
            cin >> A[i][j];
        }
    }
    for (int i = 0; i < m; i++)
    {
        for (int j = 0 ; j < n; j++)
        {
            cout << A[i][j] << "\t";
        }
        cout << endl;
    }

    if (m != n)
    {
        cout << "Khong phai ma tran vuong!!!" << endl;
        return 0;
    }

    bool duongcheo = true, tamgiactren = true, tamgiacduoi = true, doixung = true;

    // Kiểm tra các loại ma trận
    for (int i = 0; i < m; i++)
    {
        for (int j = 0; j < n; j++)
        {
            if (i != j && A[i][j] != 0)
                duongcheo = false;

            if (i > j && A[i][j] != 0)
                tamgiactren = false;

            if (i < j && A[i][j] != 0)
                tamgiacduoi = false;

            if (A[i][j] != A[j][i])
                doixung = false;
        }
    }

    if (duongcheo)
        cout << "Ma tran nay la ma tran duong cheo." << endl;
    else if (tamgiactren)
        cout << "Ma tran nay la ma tran tam giac tren." << endl;
    else if (tamgiacduoi)
        cout << "Ma tran nay la ma tran tam giac duoi." << endl;
    else if (doixung)
        cout << "Ma tran nay la ma tran doi xung." << endl;
    else
        cout << "Ma tran nay khong thuoc cac loai tren." << endl;

    return 0;
}